### PR TITLE
Select opengl

### PR DESCRIPTION
**Pull request by pchampio**
Select the OpenGL version to use.

wraps the go-gl/gl lib in a internal package with build constrains to
let the user choose which go-gl/gl version he desire.

support 3.3(default) and none.

When the build tag is set to none, no dependency to go-gl/gl is
required, but texture plugins will not be supported.

related hover PR: https://github.com/go-flutter-desktop/hover/pull/24

### CODE REVIEW

File: internal/opengl/opengl.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+// The default version (3.3) of OpenGL used by go-flutter.
+// If you want to support other version, copy/pase this file, change the import
+// statement, add builds constraints and open a PR.
+
+import (
+	"unsafe"
+
+	"github.com/go-gl/gl/v3.3-core/gl"
+	"github.com/go-gl/glfw/v3.2/glfw"
```
Comment (GeertJohan) [inline internal/opengl/opengl.go]: 3.2? Not 3.3?

**Comment by GeertJohan** [comment]
One comment, otherwise LGTM

File: internal/opengl/opengl.go (hunk 1 above)
Comment (Drakirus) [reply internal/opengl/opengl.go]: I've chosen 3.3 was based on [this](https://www.reddit.com/r/opengl/comments/51htbk/what_is_the_best_opengl_version_to_start_with/d7c7erf?utm_source=share&utm_medium=web2x) comment from a Reddit post.

> Short answer: a core OpenGL 3.3 profile. You'll get wide support across all major operating systems with this.
> OS X and Mesa will be your lowest-end target APIs, and they bottom out at GL 3.3 for the oldest and gimpiest. GL 3.3 is a "modern" API meaning that a lot of what you do here will be comparable to other 3D APIs in concept and execution.


The post has 3years but I think it is still relevant as OpenGL moves very slowly.

I'm NOT familiar with OpenGL, if you think 3.2 is a better choice, I will downgrade. (Not comfortable enough to have any opinion)

**Comment by GeertJohan** [comment]
LGTM, but needs to be merged together with the hover side I guess?

**Pull request merged**